### DISCUSSION TITLE
font awesome icon does not stay in the color defined in CSS when exporting cshtml page to pdf

### DISCUSSION THREAD

**Discussion by juliermefelix** [serenity-is/Serenity#6737] [Help] [answered]
I have a report in a cshtml page that I export to PDF, but when I add an icon through font awesome and I configure the CSS as instructed to present the icon in white color, however when generating the PDF the icon is generated in black color.
I did the test by generating the same page but for html and the icon appears in white as it should be.
Where do I fix this problem?

export image in pdf
![image](https://user-images.githubusercontent.com/7516443/227516795-1c159413-23d2-4498-87eb-61f847ad81cd.png)

export image in html
![image](https://user-images.githubusercontent.com/7516443/227517026-4f1a3c8e-b41e-4797-85bd-651b7c6e09d3.png)

**Comment by muhammedyaman** [comment, marked as answer]
How are you giving the white color?

**Reply by juliermefelix** [reply]
I've tried through the css and directly in the tag
css
 .fa-whatsapp {
            margin-right: 10px;
            color:#ffffff !important;
        }

directly in the tag
`<i` class="fa-brands fa-whatsapp" style="color: #ffffff;"></i>`

**Reply by juliermefelix** [reply]
Probably the problem is not in assigning the colors, but in some class when it generates in pdf it is overlapping the color, because when I change the output to html in the ReportHelpers the color is presented as white.

part of the ReportHelpers class where I modify the output from pdf to html
`export function execute(options: ReportExecuteOptions) {
            var opt = options.getParams ? options.getParams() : options.params;

            Q.postToUrl({
                url: '~/Report/' + (options.download ? 'Download' : 'Render'),
                params: {
                    key: options.reportKey,
                    ext: Q.coalesce(options.extension, 'pdf'),
                    opt: opt ? $.toJSON(opt) : ''
                },
               
                target: Q.coalesce(options.target, '_blank')
            });
        }`

![image](https://user-images.githubusercontent.com/7516443/229140636-08ede584-f1d2-49eb-b054-5c135db0718f.png)

output to html
![image](https://user-images.githubusercontent.com/7516443/229140793-a58ec6be-04dc-45e5-9a54-950fe5e7d4ee.png)

**Reply by muhammedyaman** [reply]
There must be a xyxReport.cs file. Did you check your Customize method override? I am not using Serenity's default wkhtmltopdf based structure anymore so I don't have a sample or xyzReport.cs file. I am guessing the diff should be in that Customize method. There was some options for converting to pdf.

**Reply by juliermefelix** [reply]
Thanks for the direction, I changed the option "UsePrintMediaType" to "false" in xxreport.cs in the Customize method  of the pdf and now I got the expected result.
![image](https://user-images.githubusercontent.com/7516443/229482523-53b47918-94e9-40f5-b27e-ad5ed152d8aa.png)

pdf
![image](https://user-images.githubusercontent.com/7516443/229482688-46665005-f0c3-4713-8598-8def69792b15.png)

**Reply by muhammedyaman** [reply]
I'm glad its resolved. You're welcome!
